timelock-agent/src/main/java/com/palantir/timelock/paxos/TimeLockAgent.java

```diff
@@ -171,6 +179,23 @@ private void createAndRegisterResources() {
                 this::createInvalidatingTimeLockServices,
                 Suppliers.compose(TimeLockRuntimeConfiguration::maxNumberOfClients, runtime::get));
 
+        if (install.paxos().leaderMode() == PaxosLeaderMode.LEADER_PER_CLIENT) {
+            LocalPaxosComponents paxosComponents = Preconditions.checkNotNull(
+                    paxosResources.leadershipBatchComponents().get(PaxosUseCase.LEADER_FOR_EACH_CLIENT),
+                    "Timelock setup has gone awfully wrong, this is a bug.");
+
```

Comment:
Could this check be moved to a public method in  paxosResources?

Reply:
I don't believe so, in this portion of the code, I need to get a particular set of batch components, whereas in `PaxosResources` it will populate whatever config has been setup. So in any case, if I fetch from the map, I need to ensure that the entry exists.

Alternatively, I wanted this additiong to go inside `PaxosResources(Factory)` but that involved opening up too many holes/trying to satisfy dependency injection, so I gave up (circular dependency exists).